pkg/cmd/cmd.go

```diff
@@ -115,10 +134,34 @@ func addFlags(opt *option, f *pflag.FlagSet) {
 	f.IntVar(&opt.Timeout, FlagTimeout, defaultTimeout, "Timeout for running each action")
 	f.IntVar(&opt.RegistrySize, FlagRegistrySize, defaultRegistrySize, "Cache size for filters and actions")
 	f.StringVar(&k8sresourcewatcher.MultiClusterConfigType, "multi-cluster-config-type", k8sresourcewatcher.TypeClusterGateway, "Multi-cluster config type, supported types: cluster-gateway, cluster-gateway-kubeconfig")
+	f.BoolVar(&enableLeaderElection, FlagLeaderElect, false, "Enable leader election for kube-trigger. Enabling this will ensure there is only one active kube-trigger.")
+	f.DurationVar(&leaseDuration, FlagLeaderElectionLeaseDuration, defaultLeaseDuration, "The duration that non-leader candidates will wait to force acquire leadership.")
+	f.DurationVar(&renewDeadline, FlagLeaderElectionRenewDeadline, defaultRenewDeadline, "The duration that the acting controlplane will retry refreshing leadership before giving up.")
+	f.DurationVar(&retryPeriod, FlagLeaderElectionRetryPeriod, defaultRetryPeriod, "The duration the LeaderElector clients should wait between tries of actions.")
 }
 
 func runCli(cmd *cobra.Command, args []string) error {
 	ctx := cmd.Context()
```

Comment:
Should `ctx` be cancelled on return? (it should only matter if leader election is disabled). From my understanding, 

If leader election is disabled, the `ctx` in `r.start(ctx)` call is not cancelled on termination, so the runner may not be told to shutdown.

https://github.com/kubevela/kube-trigger/blob/c337b56e3defd55fd68b05280b582573b21ad908/pkg/cmd/cmd.go#L258

If leader election is enabled, then this should be fine, since `ctx` will cancel once leader stepped down.

Reply:
sure, I will fix it